clients/ts/signalr/src/ServerSentEventsTransport.ts

```diff
@@ -32,7 +36,7 @@ export class ServerSentEventsTransport implements ITransport {
         Arg.isRequired(transferFormat, "transferFormat");
         Arg.isIn(transferFormat, TransferFormat, "transferFormat");
 
-        if (typeof (EventSource) === "undefined") {
+        if (!this.eventSourceConstructor || this.eventSourceConstructor === undefined) {
```

Comment:
`!this.eventSourceConstructor` does not also cover `undefined`?

Reply:
Why not do this check in the constructor? Then eventSourceConstructor field doesn't need to be optional.

Reply:
Actually you could do it before the constructor is called in `HttpConnection`.

Ditto parameter for `WebSocketTransport`.

Reply:
We just have to make sure the fallback behavior works the same when we do that.

Reply:
Fallback works, we check the constructors in `resolveTransport` and skip them if they don't exist